conjure-java-core/src/main/java/com/palantir/conjure/java/services/UndertowServiceHandlerGenerator.java

```diff
@@ -487,27 +491,89 @@ private CodeBlock decodePlainParameterCodeBlock(Type type, TypeMapper typeMapper
                     paramId
             );
         }
-        if (type.accept(TypeVisitor.IS_OPTIONAL)
-                && type.accept(TypeVisitor.OPTIONAL).getItemType().accept(TypeVisitor.IS_EXTERNAL)) {
-            return CodeBlocks.statement(
-                    "$1T $2N = $4T.deserializeOptionalString($5N.get($6S)).map($3T::valueOf)",
-                    typeMapper.getClassName(type),
-                    resultVarName,
-                    typeMapper.getClassName(type.accept(TypeVisitor.OPTIONAL).getItemType()),
-                    StringDeserializers.class,
-                    paramsVarName,
-                    paramId
-            );
-        }
-        return CodeBlocks.statement(
+        Optional<CodeBlock> complexDeserializer = getComplexTypeDeserializer(
```

Comment:
For once, I would argue that it might be more readable using an if-else statement:

`if (isPrimitive) ... else ...`

Reply:
ha, updated